File: apps/web/src/types/asset.ts
export interface AssetType {
	id: number;
	name_th: string;
	name_en?: string;
}

export interface Asset {
	id: number;
	asset_code: string;
	name_th: string | null;
	name_en: string | null;
	asset_type_id: number | null;
	price: number | null;
	bedrooms: number | null;
	bathrooms: number | null;
	description_th: string | null;
	description_en: string | null;
	location_latitude: number | null;
	location_longitude: number | null;
	images_main_id: number | null;
}

export interface AssetListResponse {
	items: Asset[];
	total: number;
}

export interface SearchFilter {
	query: string;
	price_max?: number;
	radius_km?: number;
	location_keyword?: string;
}

export interface SearchRequest {
	query_text: string;
	filters: {
		asset_type_id?: number[];
		price_min?: number;
		price_max?: number;
		bedrooms_min?: number;
	};
	pagination: {
		page: number;
		page_size: number;
	};
}

export interface AssetResult {
	id: number;
	asset_code: string;
	name_th: string | null;
	price: number | null;
	image_url: string;
	location_latitude: number | null;
	location_longitude: number | null;
}

export interface SearchResponse {
	results: AssetResult[];
	total_pages: number;
}

File: apps/api/app/schemas/asset.py
"""Schemas for Asset CRUD operations."""


from pydantic import BaseModel, Field


class AssetBase(BaseModel):
    asset_code: str = Field(min_length=1)
    name_th: str | None = None
    name_en: str | None = None
    asset_type_id: int | None = None
    price: float | None = None
    bedrooms: int | None = None
    bathrooms: int | None = None
    description_th: str | None = None
    description_en: str | None = None
    location_latitude: float | None = None
    location_longitude: float | None = None
    images_main_id: int | None = None


class AssetCreate(AssetBase):
    pass


class AssetUpdate(BaseModel):
    asset_code: str | None = None
    name_th: str | None = None
    name_en: str | None = None
    asset_type_id: int | None = None
    price: float | None = None
    bedrooms: int | None = None
    bathrooms: int | None = None
    description_th: str | None = None
    description_en: str | None = None
    location_latitude: float | None = None
    location_longitude: float | None = None
    images_main_id: int | None = None
    skip_embedding: bool = False


class AssetResponse(AssetBase):
    id: int

    model_config = {"from_attributes": True}


class AssetListResponse(BaseModel):
    items: list[AssetResponse]
    total: int


class AssetTypeResponse(BaseModel):
    id: int
    name_th: str
    name_en: str

    model_config = {"from_attributes": True}


class AssetTypeListResponse(BaseModel):
    items: list[AssetTypeResponse]
